### DISCUSSION TITLE
Support associated types with bounds

### DISCUSSION THREAD

**Discussion by marshtompsxd** [verus-lang/verus#674] [General]
It would be great if we can have associated types with bounds.
Current it reports the error:
```
error: The verifier does not yet support the following Rust feature: unsupported item
  --> reconciler/spec/mod.rs:34:1
   |
34 | / pub trait SomeTrait {
35 | |     type A: OtherTrait;
36 | |     type B;
37 | | }
   | |_^
```

**Comment by marshtompsxd** [comment]
A common use case of this is to simplify lemmas. Currently, we have to carry multiple bounded generic types in each lemma but with this support we will be able to fold them into one (with associated types). Besides, without this support we find it difficult to generate some proof obligations in our framework.

**Comment by utaal-b** [comment]
This is now supported, see https://github.com/verus-lang/verus/pull/856.
